staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt-config.go

```diff
@@ -0,0 +1,276 @@
+	reqMgmt.configQueue = workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(200*time.Millisecond, 8*time.Hour), "req_mgmt_config_queue")
+	fci := informerFactory.Flowcontrol().V1alpha1()
+	pli := fci.PriorityLevelConfigurations()
+	fsi := fci.FlowSchemas()
+	reqMgmt.plLister = pli.Lister()
+	reqMgmt.plInformerSynced = pli.Informer().HasSynced
+	reqMgmt.fsLister = fsi.Lister()
+	reqMgmt.fsInformerSynced = fsi.Informer().HasSynced
+	pli.Informer().AddEventHandler(reqMgmt)
+	fsi.Informer().AddEventHandler(reqMgmt)
+}
+
+func (reqMgmt *requestManagementSystem) triggerReload() {
+	klog.V(7).Info("triggered request-management system reloading")
+	reqMgmt.configQueue.Add(0)
+}
+
+// OnAdd handles notification from an informer of object creation
+func (reqMgmt *requestManagementSystem) OnAdd(obj interface{}) {
+	reqMgmt.triggerReload()
+}
+
+// OnUpdate handles notification from an informer of object update
+func (reqMgmt *requestManagementSystem) OnUpdate(oldObj, newObj interface{}) {
+	reqMgmt.triggerReload()
+}
+
+// OnDelete handles notification from an informer of object deletion
+func (reqMgmt *requestManagementSystem) OnDelete(obj interface{}) {
+	reqMgmt.triggerReload()
+}
+
+func (reqMgmt *requestManagementSystem) Run(stopCh <-chan struct{}) error {
+	defer reqMgmt.configQueue.ShutDown()
+	klog.Info("Starting reqmgmt config controller")
+	if ok := cache.WaitForCacheSync(stopCh, reqMgmt.readyFunc, reqMgmt.plInformerSynced, reqMgmt.fsInformerSynced); !ok {
+		return fmt.Errorf("Never achieved initial sync")
+	}
+	klog.Info("Running reqmgmt config worker")
+	wait.Until(reqMgmt.runWorker, time.Second, stopCh)
+	klog.Info("Shutting down reqmgmt config worker")
+	return nil
+}
+
+func (reqMgmt *requestManagementSystem) runWorker() {
+	for reqMgmt.processNextWorkItem() {
+	}
+}
+
+func (reqMgmt *requestManagementSystem) processNextWorkItem() bool {
+	obj, shutdown := reqMgmt.configQueue.Get()
+	if shutdown {
+		return false
+	}
+
+	func(obj interface{}) {
+		defer reqMgmt.configQueue.Done(obj)
+		if !reqMgmt.syncOne() {
+			reqMgmt.configQueue.AddRateLimited(obj)
+		} else {
+			reqMgmt.configQueue.Forget(obj)
+		}
+	}(obj)
+
+	return true
+}
+
+// syncOne attempts to sync all the config for the reqmgmt filter.  It
+// either succeeds and returns `true` or logs an error and returns
+// `false`.
+func (reqMgmt *requestManagementSystem) syncOne() bool {
+	all := labels.Everything()
+	newPLs, err := reqMgmt.plLister.List(all)
+	if err != nil {
+		klog.Errorf("Unable to list PriorityLevelConfiguration objects: %s", err.Error())
+		return false
+	}
+	newFSs, err := reqMgmt.fsLister.List(all)
+	if err != nil {
+		klog.Errorf("Unable to list FlowSchema objects: %s", err.Error())
+		return false
+	}
+	reqMgmt.digestConfigObjects(newPLs, newFSs)
+	return true
+}
+
+func (reqMgmt *requestManagementSystem) digestConfigObjects(newPLs []*rmtypesv1a1.PriorityLevelConfiguration, newFSs []*rmtypesv1a1.FlowSchema) {
```

Comment:
BTW, the logic here does not properly handle the case of changing the Exempt bit.  See my PRs to code and KEP.

Reply:
i dont think we need global-default given that we can actually use (authenticated+unauthenticated) to represent (all request)..

Reply:
That's a different matter; let's discuss it in a place dedicated to that purpose, perhaps it in #82196 or https://github.com/kubernetes/enhancements/pull/1223 ?